app/ota_client.py

```diff
@@ -50,73 +49,91 @@ def verify_file(filename: Path, filehash: str, filesize) -> bool:
-    quoted_path = urllib.parse.quote(path)
-    url = urllib.parse.urljoin(url_base, quoted_path)
-
-    error_count = 0
-    last_error = ""
-
-    def retry_if_possible(e):
-        nonlocal error_count
-        nonlocal last_error
-        error_count += 1
-        do_retry = True
-        if isinstance(e, requests.exceptions.Timeout):
-            last_error = f"requests timeout: {e},{url=} ({error_count})"
-        elif isinstance(e, requests.exceptions.ConnectionError):
-            last_error = f"requests connection error: {e},{url=} ({error_count})"
-        elif isinstance(e, requests.exceptions.ChunkedEncodingError):
-            last_error = f"requests ChunkedEncodingError: {url=} ({error_count})"
-        elif isinstance(e, RequestException):
-            status_code = getattr(e.response, "status_code", None)
-            last_error = f"requests error: {status_code=},{url=} ({error_count})"
-            if isinstance(status_code, int) and status_code // 100 == 4:
-                do_retry = False
-        else:
-            last_error = f"requests error unknown: {e},{url=}, ({error_count})"
-        logger.warning(last_error)
-        return do_retry
-
-    @retry(
-        stop_max_attempt_number=5,
-        retry_on_exception=retry_if_possible,
-        wait_exponential_multiplier=1_000,
-        wait_exponential_max=10_000,
-    )
-    def _requests_get():
-        CHUNK_SIZE = 4096
-        headers = {}
-        headers["Accept-encording"] = "gzip"
-        response = requests.get(
-            url, headers=headers, cookies=cookies, timeout=10, stream=True
-        )
-        response.raise_for_status()
+class Downloader:
+    CHUNK_SIZE = 2 * 1024 * 1024  # 2MiB
+    RETRY_COUNT = 5
 
-        with open(dst, "wb") as f:
-            m = sha256()
-            total_length = response.headers.get("content-length")
-            if total_length is None:
-                m.update(response.content)
-                f.write(response.content)
-            else:
-                for data in response.iter_content(chunk_size=CHUNK_SIZE):
-                    m.update(data)
-                    f.write(data)
-        return m.hexdigest()
+    def __init__(self, *, proxy: str):
+        from requests.adapters import HTTPAdapter
+        from urllib3.util.retry import Retry
+
+        # base session
+        session = requests.Session()
 
-    try:
-        calc_digest = _requests_get()
-    except Exception:
-        logger.exception("requests_get")
-        raise OtaErrorRecoverable(last_error)
+        # configure proxy
+        proxies = {"http": proxy, "https": ""}
+        session.proxies.update(proxies)
 
-    if digest and digest != calc_digest:
-        raise OtaErrorRecoverable(
-            f"hash error: act={calc_digest}, exp={digest}, {url=}"
+        # init retry mechanism
+        retry_strategy = Retry(
+            total=self.RETRY_COUNT,
+            raise_on_status=True,
+            backoff_factor=0.1,
```

Comment:
To keep previsous implementation, I'd like to use this parameter:
```suggestion
            backoff_factor=1,
            backoff_max=10,            
```